lib/mwUtil.js

```diff
@@ -412,13 +412,19 @@ mwUtil.isSelfRedirect = (req, location) => {
 
 /**
  * Fetches the summary content from the provided URI
- * @param  {HyperSwitch}  hyper HyperSwitch context
- * @param  {URI}          uri   summary URI
- * @return {string}             summary body content
+ * @param  {HyperSwitch}  hyper     HyperSwitch context
+ * @param  {URI}          uri       summary URI
+ * @param  {headers}      headers    req header
+ * @return {string}                 summary body content
  */
-mwUtil.fetchSummary = (hyper, uri) => {
-    return hyper.get({ uri })
+mwUtil.fetchSummary = (mRes, hyper, uri, headers = undefined) => {
+    return hyper.get({ uri, headers })
     .then((res) => {
+        // Assign content-language and vary header based on one of summary responses
+        if (mRes.headers && !mRes.headers['content-language']) {
+            mRes.headers['content-language'] = res.headers && res.headers['content-language']
+            mwUtil.addVaryHeader(res, 'accept-language');
```

Comment:
Why are you modifying both `res` and `mRes` headers?

Reply:
Nice catch, thanks.